### PR TITLE
SALTO-5335: add conditionGroups to transition condition

### PR DESCRIPTION
**Pull request by shayc331**
_add conditionGroups to transition condition_

---

_Additional context for reviewer_
The nacl looks like this: 
Screen Shot 2024-01-25 at 17 20 40
checkout the API [documentation](https://developer.atlassian.com/cloud/jira/platform/rest/v3/api-group-workflows/#api-rest-api-3-workflows-update-post) for more info

* more info in the [ticket](https://salto-io.atlassian.net/browse/SALTO-5335)

---
_Release Notes_: 
_None_

---
_User Notifications_: 
_None_

### CODE REVIEW

File: packages/jira-adapter/src/filters/workflowV2/workflow_filter.ts [hunk 1]
```diff
@@ -344,6 +344,21 @@ const convertParametersFieldsToString = (
...
 
+// Jira has a bug that causes conditionGroups to be required in the deployment requests
+// We should remove this once the bug is fixed - https://jira.atlassian.com/browse/JRACLOUD-82794
+const insertConditionGroups:WalkOnFunc = ({ value, path }): WALK_NEXT_STEP => {
+  if (_.isPlainObject(value) && value.operation && value.conditionGroups === undefined) {
+    value.conditionGroups = []
+  }
+  if (isInstanceElement(value)
+  || path.name === 'transitions'
+  || path.name === 'conditions'
```
Comment (dantal4) [inline packages/jira-adapter/src/filters/workflowV2/workflow_filter.ts:355]: when does this happen? where `path.name === 'conditions'` you already getting to the first condition in line 350, right?

File: packages/jira-adapter/src/filters/workflowV2/workflow_filter.ts [hunk 2]
```diff
@@ -344,6 +344,21 @@ const convertParametersFieldsToString = (
     })
 }
 
+// Jira has a bug that causes conditionGroups to be required in the deployment requests
+// We should remove this once the bug is fixed - https://jira.atlassian.com/browse/JRACLOUD-82794
+const insertConditionGroups:WalkOnFunc = ({ value, path }): WALK_NEXT_STEP => {
+  if (_.isPlainObject(value) && value.operation && value.conditionGroups === undefined) {
```
Comment (dantal4) [inline packages/jira-adapter/src/filters/workflowV2/workflow_filter.ts:350]: can `value.operation` be defined but falsy (`false`/`0`/`''`)

Comment (dantal4) [reply packages/jira-adapter/src/filters/workflowV2/workflow_filter.ts:350]: I generally use `var !== undefined`, but in this specific case I get it that it's not needed 👍

File: packages/jira-adapter/src/filters/workflowV2/workflow_filter.ts (hunk 1 above)
Comment (shayc331) [reply packages/jira-adapter/src/filters/workflowV2/workflow_filter.ts:355]: the path is like this: `conditions.conditions/operation/conditionGroups`
Screen Shot 2024-01-25 at 16 19 52
so I need to enter the first `conditions`

Comment (dantal4) [reply packages/jira-adapter/src/filters/workflowV2/workflow_filter.ts:355]: you enter it here - `_.isPlainObject(value) && value.operation && value.conditionGroups === undefined`, no?

Comment (shayc331) [reply packages/jira-adapter/src/filters/workflowV2/workflow_filter.ts:355]: yes, but to the nested `conditions`.
I need to enter the first `conditions` as well

Comment (dantal4) [reply packages/jira-adapter/src/filters/workflowV2/workflow_filter.ts]: Ok, I see that now the test have the nested condition too, so I understand it now 🙏 
can you show in the description of the PR how the nacl looks?

Comment (shayc331) [reply packages/jira-adapter/src/filters/workflowV2/workflow_filter.ts]: sure, I added it to the description. If something is still not clear please let me know

**Review by dantal4 — APPROVED** [review]
💯

**Pull request merged**